src/utils/AuthProvider.tsx

```diff
@@ -101,6 +102,15 @@ export default function AuthProvider({
         options,
       }); // will trigger onAuthStateChange to update the session
 
+      // This code is largely taken from https://github.com/orgs/supabase/discussions/1282#discussioncomment-5230475
+
+      // User exists, but is fake. See https://supabase.com/docs/reference/javascript/auth-signup
+      if (await checkEmailExists(email)) {
```

Comment:
nit: might be cleaner if we store the awaited result in a variable first and use it here